### ISSUE TITLE
TFX Command-line pipeline update command only works on machine created

### ISSUE THREAD

**Issue by kyledef** [tensorflow/tfx#2420]
Our team was successfully able to create and update a pipeline with the commands `tfx pipeline create ....` and `tfx pipeline update ...` on the same machine. However, when we tried to update a pipeline from another user's machine with the same endpoint we get the error `Pipeline "PIPELINE_NAME" does not exist.`

We can see the pipeline when we use the list command `tfx pipeline list --endpoint $ENDPOINT` the pipeline is correctly listed, but its we still cannot update it.

**Comment by jiyongjung0** [comment]
Thank you for the report.

It is because current CLI implementation stores pipeline information in local environment. It would be better not to store such information and read all required information from KFP api server, but it might take some time to improve.

As a work-around you can copy the content of `~/kubeflow/${PIPELINE_NAME}/` to the new machine. Note that files in this directory will be changed every time the pipeline is updated. Good news is that the path to store this pipeline information can be customized with `KUBEFLOW_HOME` environment variable. If you set KUBEFLOW_HOME to the directory path under your source code control repository, you will be able to share the required files.

You can find [related code here](https://github.com/tensorflow/tfx/blob/ff6917997340401570d05a4d3ebd6e8ab5760495/tfx/tools/cli/handler/kubeflow_handler.py#L268).

**Comment by kyledef** [comment]
Thanks for the response, we'll go the route of incorporating the home environment into the project. Seem's a bit messy, but its a suitable work around.

**Comment by thomas-vl** [comment]
@jiyongjung0 this means we cant deploy pipelines from serverless build solutions.
Are there any plans to not require the kubeflow home directory for an update?

**Comment by jiyongjung0** [comment]
@thomas-vl Thank you for you interest. We understand that this issue is a non-trivial blocker in user experiences, but couldn't have time to look into it. I think that we just need to send a few more requests rather than storing information in local storage. I'll do some experiments and come back with more concrete plan in a few days.

Thanks!

**Comment by jiyongjung0** [comment]
Thank you for patience. I've changed kubeflow usage not to depend local filesystem at all and all the above hack is not needed any more. (See the merged PR above for detail.)

This change will be included in 0.27.0 release which will happen in about a month. But you can also try it using our nightly build using following command.

```
pip install -i https://pypi-nightly.tensorflow.org/simple tfx
```

**Comment by jiyongjung0** [comment]
TFX 0.27.0 was released and now you can use the released package.
